biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -461,7 +461,14 @@ private TaggedData doCached0() throws Exception {
 				//
 
 				if ("file".equalsIgnoreCase(url.getProtocol())) {
-					File sourceFile = new File(uri);
+					// HTTP clients are required to strip fragments before
+					// querying so we should remove any fragment here
+					File sourceFile;
+					if (uri.getFragment() == null) {
+						sourceFile = new File(uri);
+					} else {
+						sourceFile = new File(uri.getPath());
+					}
```

Comment:
I don't think so. The URL we generate is still valid and conforms to the URL specification. I think that the change here is an indication that you need to be careful when "going native" with a URL. If the code here just used `openStream()` it would work fine, as would just `new File(uri.getPath())`. Instead the code assumes that the URL is convertible into a URI without changes, and that the URI has no fragment or query data.